tensorflow/compiler/xla/service/mlir_gpu/BUILD

```diff
@@ -113,6 +113,7 @@ cc_library(
         "//tensorflow/compiler/xla/service/gpu:ir_emission_utils",
         "//tensorflow/compiler/xla/service/gpu:nvptx_compiler_impl",
         "//tensorflow/compiler/xla/service/gpu:launch_dimensions",
+        "//tensorflow/compiler/xla/service/gpu:nvptx_compiler",
```

Comment:
This extra dependency breaks `tensorflow/compiler/xla/service/mlir_gpu/tests`. The linking step will register a compiler and these tests fail with `Check failed: factories->find(platform_id) == factories->end() Compiler factory already registered for platform`.
Please consider moving `WarnIfBadDriverJITVersion` into a different place. I believe asm_compiler.cc would work.

Reply:
I amended the commit to just remove this line. This fix the mlir_gpu tests. `//tensorflow/compiler/xla/service/gpu:nvptx_compiler_impl` is already included and it is enough. As it was already included, I do not see value in moving that function.

Reply:
You are right, the dependency to nvptx_compiler_impl is enough and the other one was redundant/wrong.